### PR TITLE
fix: logged in as none and reprocess finetune fixes

### PR DESCRIPTION
**Pull request by elboy3**
## Shortcut

https://app.shortcut.com/galileo/story/11305/dq-logged-in-as-none?vc_group_by=day&ct_workflow=all&cf_workflow=500000005

https://app.shortcut.com/galileo/story/11306/dq-reprocess-broken-for-finetune?vc_group_by=day&ct_workflow=all&cf_workflow=500000005

Tested login wiht a few flows and was able to reproduce and fix:
```
>>> dq.login()
📡 https://console.dev.rungalileo.io
🔭 Logging you into Galileo

🚀 You're logged in to Galileo as galileo@rungalileo.io!
Use logout() if you want to change users
```

Tested reprocess locally and it worked like a charm 
```
>>> reprocess_run("Seq2Seq_DecoderOnly_Log_Logprobs", "2023-11-13_9")
All alerts removed for run Seq2Seq_DecoderOnly_Log_Logprobs/2023-11-13_9
Job default successfully resubmitted. New results will be available soon at https://console.dev.rungalileo.io/insights/4551e611-011a-4004-a8ba-b3113980f2ca/2f2fdd25-68ae-4be0-be94-ef62590a7345?split=training&taskType=8
Waiting for job (you can safely close this window)...
        Downloading all embedding files for this run
        [training] 👀 Looking for data anomalies
Done! Job finished with status completed
{'project_id': '4551e611-011a-4004-a8ba-b3113980f2ca', 'run_id': '2f2fdd25-68ae-4be0-be94-ef62590a7345', 'job_id': '548062a2-0abd-4dc9-bffa-87b20ae7ac85', 'job_name': 'default', 'task_type': 8, 'labels': [], 'ner_labels': [], 'tasks': None, 'non_inference_logged': False, 'migration_name': None, 'xray': True, 'process_existing_inference_runs': False, 'feature_names': [], 'prompt_dataset_id': None, 'prompt_template_version_id': None, 'monitor_batch_id': None, 'prompt_settings': None, 'prompt_scorers_configuration': None, 'prompt_registered_scorers_configuration': None, 'prompt_scorer_settings': None, 'message': 'Processing job!', 'link': '/insights/4551e611-011a-4004-a8ba-b3113980f2ca/2f2fdd25-68ae-4be0-be94-ef62590a7345?split=training&taskType=8'}
```

### CODE REVIEW

File: dataquality/core/auth.py [hunk 1]
```diff
@@ -49,27 +49,23 @@ def login() -> None:
...
-        return
-
+    valid_current_user = api_client.valid_current_user()
     print(f"📡 {config.api_url.replace('api','console')}")
     print("🔭 Logging you into Galileo\n")
 
-    _auth = _Auth()
-    if os.getenv("GALILEO_USERNAME") and os.getenv("GALILEO_PASSWORD"):
-        _auth.login_with_env_vars()
-    else:
```
Comment (elboy3) [inline dataquality/core/auth.py:64]: can simplify process since we refresh token in valid_current_user

File: dataquality/internal.py [hunk 2]
```diff
@@ -27,52 +27,24 @@ def reprocess_run(
...
-        ner_labels=ner_labels,
-        xray=alerts,
-        # We set the job name to inference and non_inference_logged to True because
-        # This will force the server to first reprocess the non-inference splits,
-        # and then reprocess all of the inference splits. If there are no inference
-        # splits, this will still work as expected, inference will just be skipped
-        job_name="inference",
-        process_existing_inference_runs=True,
-        non_inference_logged=True,
-    )
```
Comment (elboy3) [inline dataquality/internal.py:73]: by using job data this is way cleaner now

File: dataquality/internal.py [hunk 3]
```diff
@@ -27,52 +27,24 @@ def reprocess_run(
...
+
+    job_data: Dict = job["request_data"]
+    # We need to remove the job_id from the job_data, otherwise the server will
```
Comment (ShuaiShao93) [inline dataquality/internal.py:44]: Incomplete comment?

Comment (elboy3) [reply dataquality/internal.py]: thanks, fixed

**Review by ShuaiShao93 — APPROVED** [review]
Thanks for the quick fix!

**Pull request merged**